packages/kit/src/utils/array.js

```diff
@@ -0,0 +1,11 @@
+/**
+ * Removes falsy values from an array.
+ *
+ * @function
+ * @template T
+ * @param {Array<T>} arr
+ * @returns {Array<NonNullable<T>>}
```

Comment:
We can simplify this a bit since the value is inferred from the return statement

```suggestion
 * @template T
 * @param {Array<T>} arr
```